debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -361,12 +361,21 @@ protected Object convertTimestampWithZone(Column column, Field fieldDefn, Object
 
 
     protected Object convertTimestampWithZoneToEpochMillisAsDate(Column column, Field fieldDefn, Object data) {
-        return convertValue(column, fieldDefn, data, fallbackTimestampWithTimeZone, (r) -> {
-            try {
-                r.deliver(new java.util.Date(((java.time.ZonedDateTime) data).toInstant().toEpochMilli()));
-            }
-            catch (IllegalArgumentException e) {
+    return convertValue(
+        column,
+        fieldDefn,
+        data,
+        fallbackTimestampWithTimeZone,
+        (r) -> {
+          try {
+            if (data instanceof java.time.ZonedDateTime) {
+              r.deliver(
+                  new java.util.Date(((java.time.ZonedDateTime) data).toInstant().toEpochMilli()));
+            } else if(data instanceof java.sql.Timestamp){
+                r.deliver(new java.util.Date(Timestamp.toEpochMillis(data, adjuster)));
             }
```

Comment:
should we do a log in else to understand if any use case is there?

Reply:
dine